src/components/language/language.service.ts

```diff
@@ -537,46 +546,36 @@ export class LanguageService {
     session: ISession
   ): Promise<LanguageListOutput> {
     const label = 'Language';
-    // const baseNodeMetaProps = ['id', 'createdAt'];
-    // const unsecureProps = [''];
-    const secureProps = [
-      'name',
-      'displayName',
-      'isDialect',
-      'populationOverride',
-      'registryOfDialectsCode',
-      'leastOfThese',
-      'leastOfTheseReason',
-      'displayNamePronunciation',
-      'sensitivity',
-      'sponsorDate',
-    ];
-
-    const query = this.db.query().call(matchRequestingUser, session);
-    // .call(matchUserPermissions, 'Language');
-
-    if (filter.name) {
-      query.call(filterByString, label, 'name', filter.name);
-    }
-
-    query.match([node('node', 'Language', { active: true })]);
 
-    const result = await runListQuery<Node<BaseNode>>(
-      query,
-      input,
-      secureProps.includes(input.sort)
-    );
-    const items = await Promise.all(
-      result.items.map(async (item) => {
-        return await this.readOne(item.properties.id, session);
-      })
-    );
+    const query = this.db
+      .query()
+      .match([
+        requestingUser(session),
+        ...permissionsOfNode(label),
+        ...(filter.name
+          ? [
+              relation('out', '', 'name', { active: true }),
+              node('name', 'Property', { active: true }),
+            ]
+          : []),
+      ])
+      .call((q) =>
+        filter.name ? q.where({ name: { value: contains(filter.name) } }) : q
```

Comment:
I'm seeing now where we previously called `filterByString` which handled checking permissions.
https://github.com/SeedCompany/cord-api-v3/blob/f0174b0dfb5bfb2d79f9cb6845f480c515165d3f/src/core/database/query.helpers.ts#L802
I'm wondering if this is needed here. Unfortunately the two people who started this are on vacation lol.

Reply:
Ok I created an issue for Michael. We can come back to it if needed. https://github.com/SeedCompany/cord-api-v3/issues/1109